### ISSUE TITLE
L2 preserved dropout

### ISSUE THREAD

**Issue by numpee** [jihanyang/AFN#7]
Hi,
Thanks for releasing the code for your work.

One quick question: In section 3.2 of your ICCV paper, you wrote about an L2-preserved dropout operation (to meet the adaptive L2 feature norm goal). However, I can't seem to find it in this repository. For example, the model from the [vanilla>VisDA code](https://github.com/jihanyang/AFN/blob/master/vanilla/Visda2017/SAFN/code/model/net.py) seems to use ordinary nn.Dropout - which, if I recall correctly, is the L1 preserved dropout that you mention in the paper. So is the L2 preserved dropout not used, or is it simply omitted in this release of the code?

Thanks

**Comment by jihanyang** [comment]
Hello, we implement this part in the `forward` stage of our model. For example, you can refer to the following implementation of our classifier:
https://github.com/jihanyang/AFN/blob/d96515b26223278f51f8e02477a42dbc5f13bef9/vanilla/Visda2017/SAFN/code/model/net.py#L85-L89

**Comment by numpee** [comment]
Oh I see. Didn't see that part of the code. 
So I guess if dropout probability is p (0.5 by default), then we should multiply by sqrt(1-p)
Thanks!
